docs/sphinx/examples/matrix_multiplication.rst

```diff
@@ -0,0 +1,107 @@
+..
+    Copyright (C) 2021 Dimitra Karatza
+
+    SPDX-License-Identifier: BSL-1.0
+    Distributed under the Boost Software License, Version 1.0. (See accompanying
+    file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+.. _examples_matrix_multiplication:
+
+===================
+Parallel algorithms
+===================
+
+This program will perform a matrix multiplication in parallel. The output will look something like this:
+
+.. code-block:: text
+
+   Matrix A is : 
+   4 9 6 
+   1 9 8 
+
+   Matrix B is : 
+   4 9 
+   6 1 
+   9 8 
+
+   Resultant Matrix is : 
+   124 93 
+   111 127 
+
+Setup
+=====
+
+The source code for this example can be found here:
+:download:`matrix_multiplication.cpp
+<../../examples/quickstart/matrix_multiplication.cpp>`.
+
+To compile this program, go to your |hpx| build directory (see
+:ref:`hpx_build_system` for information on configuring and building |hpx|) and
+enter:
+
+.. code-block:: shell-session
+
+   $ make examples.quickstart.matrix_multiplication
+
+To run the program type:
+
+.. code-block:: shell-session
+
+   $ ./bin/matrix_multiplication
+
+This should print:
+
+.. code-block:: text
+
+   Matrix A is : 
+   4 9 6 
+   1 9 8 
+
+   Matrix B is : 
+   4 9 
+   6 1 
+   9 8 
+   
+   Resultant Matrix is : 
+   124 93 
+   111 127 
+
+Notice that the numbers may be different because of the random initialization of the matrices.
+
+Walkthrough
+===========
+
+Now that you have compiled and run the code, let's look at how the code works.
+
+First, ``main()`` is used to initialize the runtime system and pass the command line arguments to the program. 
+``hpx::init`` calls ``hpx_main()`` after setting up HPX, which is where our program is implemented.
+
+.. literalinclude:: ../../examples/quickstart/matrix_multiplication.cpp
+   :language: c++
+   :start-after: //[mul_main
+   :end-before: //]
+
+Going on with the ``hpx_main()`` function, we can see that matrix multiplication can be done very easily. 
+
+.. literalinclude:: ../../examples/quickstart/matrix_multiplication.cpp
+   :language: c++
+   :start-after: //[mul_hpx_main
+   :end-before: //]
+
+First, the dimentions of the matrices is defined, so as that the number of columns of matrix A is equal 
```

Comment:
```suggestion
First, the dimensions of the matrices is defined, so as that the number of columns of matrix A is equal 
```

though this might need a small rewrite if the dimensions are given as command-line options.